Cannot generate declaration file when using `ESLintUtils.RuleCreator`

<!--
Please don't ignore this template.

If you ignore it, we're just going to respond asking you to fill it out, which wastes everyone's time.
The more relevant information you can include, the faster we can find the issue and fix it without asking you for more info.
-->

<!--
🚨 STOP 🚨 𝗦𝗧𝗢𝗣 🚨 𝑺𝑻𝑶𝑷 🚨

This issue template is only for problems specifically with the `@typescript-eslint/experimental-utils` package.

If you have a problem with a specific lint rule, please back out and select the `@typescript-eslint/eslint-plugin` template.
If you have a problem with the parser, please back out and select the `@typescript-eslint/parser` template.
-->

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/main/docs/linting/TROUBLESHOOTING.md) and my problem is not listed.

**Repro**

<!--
Include a ***minimal*** reproduction case.
The more irrelevant code/config you give, the harder it is for us to investigate.

Please consider creating an isolated reproduction repo to make it easy for the volunteer maintainers debug your issue.
-->

Try to transpile the following file, with `"declaration": true` in the `tsconfig.json` file.

```TS
import { ESLintUtils } from "@typescript-eslint/experimental-utils";

export const createRule = ESLintUtils.RuleCreator((s) => s);
```

**Expected Result**

<!--
What did you expect to happen?
Please be specific here - list the exact lines and messages you expect.
-->
Transpilation should succeed, and the generated declaration file should work.

**Actual Result**

<!--
What actually happened?
Please be specific here - list the exact lines and messages that caused errors
-->
Transpilation fails, with the following error log:
```
index.ts:3:14 - error TS4023: Exported variable 'createRule' has or is using name 'RuleWithMetaAndName' from external module "/PATH/TO/PROJECT/node_modules/@typescript-eslint/experimental-utils/dist/eslint-utils/RuleCreator" but cannot be named.

3 export const createRule = ESLintUtils.RuleCreator((s) => s);
               ~~~~~~~~~~


Found 1 error.
```

**Additional Info**

<!--
Did eslint throw an exception?

Please run your lint again with the --debug flag, and dump the output below.
i.e. eslint --ext ".ts,.js" src --debug
-->

**Versions**

| package                                 | version |
| --------------------------------------- | ------- |
| `@typescript-eslint/experimental-utils` | `5.4.0` |
| `TypeScript`                            | `4.5.2` |
| `node`                                  | `14.18.0` |

--- a/packages/experimental-utils/src/eslint-utils/RuleCreator.ts
+++ b/packages/experimental-utils/src/eslint-utils/RuleCreator.ts
@@ -8,12 +8,12 @@ import {
 import { applyDefault } from './applyDefault';
 
 // we automatically add the url
-type NamedCreateRuleMetaDocs = Omit<RuleMetaDataDocs, 'url'>;
-type NamedCreateRuleMeta<TMessageIds extends string> = {
+export type NamedCreateRuleMetaDocs = Omit<RuleMetaDataDocs, 'url'>;
+export type NamedCreateRuleMeta<TMessageIds extends string> = {
   docs: NamedCreateRuleMetaDocs;
 } & Omit<RuleMetaData<TMessageIds>, 'docs'>;
 
-interface CreateAndOptions<
+export interface RuleCreateAndOptions<
   TOptions extends readonly unknown[],
   TMessageIds extends string,
   TRuleListener extends RuleListener,
@@ -25,19 +25,19 @@ interface CreateAndOptions<
   defaultOptions: Readonly<TOptions>;
 }
 
-interface RuleWithMeta<
+export interface RuleWithMeta<
   TOptions extends readonly unknown[],
   TMessageIds extends string,
   TRuleListener extends RuleListener,
-> extends CreateAndOptions<TOptions, TMessageIds, TRuleListener> {
+> extends RuleCreateAndOptions<TOptions, TMessageIds, TRuleListener> {
   meta: RuleMetaData<TMessageIds>;
 }
 
-interface RuleWithMetaAndName<
+export interface RuleWithMetaAndName<
   TOptions extends readonly unknown[],
   TMessageIds extends string,
   TRuleListener extends RuleListener,
-> extends CreateAndOptions<TOptions, TMessageIds, TRuleListener> {
+> extends RuleCreateAndOptions<TOptions, TMessageIds, TRuleListener> {
   meta: NamedCreateRuleMeta<TMessageIds>;
   name: string;
 }
@@ -48,7 +48,7 @@ interface RuleWithMetaAndName<
  * @param urlCreator Creates a documentation URL for a given rule name.
  * @returns Function to create a rule with the docs URL format.
  */
-function RuleCreator(urlCreator: (ruleName: string) => string) {
+export function RuleCreator(urlCreator: (ruleName: string) => string) {
   // This function will get much easier to call when this is merged https://github.com/Microsoft/TypeScript/pull/26349
   // TODO - when the above PR lands; add type checking for the context.report `data` property
   return function createNamedRule<
@@ -106,5 +106,3 @@ function createRule<
 }
 
 RuleCreator.withoutDocs = createRule;
-
-export { RuleCreator };

